src/ros_vision_interaction/src/vision-interaction/controllers/vision_project_delegator.py

```diff
@@ -118,6 +121,17 @@ def _update_act_variables(self):
         last_5_scores.append(self._state_database.get("current eval score"))
         self._state_database.set("last 5 eval scores", last_5_scores)
 
+    def _update_reading_task_data(self):
+        last_5_scores = self._state_database.get("last 5 eval scores")
+        average_score = sum(last_5_scores)/len(last_5_scores)
+        last_score = self._state_database.get("last score")
+        if average_score - self._score_window <= last_score <= average_score + self._score_window:
```

Comment:
I think this does work here, but I am sure that this won't in all languages (like, C++). I rather have this be a function that does both checks explicitly and then you can just name it something appropriate.

Reply:
I refactored the conditionals so that I check for the 'declined' and 'improved' cases first to avoid the compound inequality.